invenio_vocabularies/contrib/names/permissions.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020-2024 CERN.
+#
+# Invenio-Vocabularies is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Vocabulary permissions."""
+
+from invenio_records_permissions.generators import AuthenticatedUser, SystemProcess
+
+from invenio_vocabularies.services.generators import Tags
+from invenio_vocabularies.services.permissions import PermissionPolicy
+
+
+class NamesPermissionPolicy(PermissionPolicy):
+    """Names permission policy."""
+
+    can_search = [
+        SystemProcess(),
+        Tags(exclude=["non-searchable"], only_authenticated=True),
+    ]
+    can_read = [SystemProcess(), AuthenticatedUser()]
+    # this permission is needed for the /api/vocabularies/ endpoint
+    can_list_vocabularies = [
+        SystemProcess(),
+        Tags(exclude=["non-searchable"], only_authenticated=True),
```

Comment:
side question, for me to understand how this works. How can I know what tags are available?

Reply:
Hmm, I think it's only documented in the RFC: https://github.com/inveniosoftware/rfcs/blob/fdb09b8b86263607364c8cc554b3a6580ccba2e2/rfcs/rdm-0077-vocabulary-harvesting.md#data-model

There is PR adding the `non-searchable`explanation, however maybe we should document htis better in the docs (?)